digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -757,25 +757,34 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
   }
 
   def getFloatingAdjacent(chainLinks: Set[Long], chainIds: Set[Long], linkId: Long, id: Long, roadNumber: Long, roadPartNumber: Long, trackCode: Int): Seq[RoadAddressLink] = {
-      val (floatings, _) = withDynTransaction {
-        RoadAddressDAO.fetchByRoadPart(roadNumber, roadPartNumber, includeFloating = true).partition(_.isFloating)
-      }
-      val historyLinks = time(logger, "Fetch floating history links") {
-        roadLinkService.getRoadLinksHistoryFromVVH(floatings.map(_.linkId).toSet)
+    val (floatings, _) = withDynTransaction {
+      RoadAddressDAO.fetchByRoadPart(roadNumber, roadPartNumber, includeFloating = true).partition(_.isFloating)
+    }
+    val historyLinks = time(logger, "Fetch floating history links") {
+      roadLinkService.getRoadLinksHistoryFromVVH(floatings.map(_.linkId).toSet)
+    }
+    if (historyLinks.nonEmpty) {
+      val historyLinkAddresses = time(logger, "Build history link addresses") {
+        historyLinks.flatMap(fh => {
+          buildFloatingRoadAddressLink(fh, floatings.filter(_.linkId == fh.linkId))
+        })
       }
-      if (historyLinks.nonEmpty) {
-        val historyLinkAddresses = time(logger, "Build history link addresses") {
-          historyLinks.flatMap(fh => {
-            buildFloatingRoadAddressLink(fh, floatings.filter(_.linkId == fh.linkId))
-          })
-        }
-        val selected = historyLinkAddresses.find(_.id == id).getOrElse(historyLinkAddresses.find(_.linkId == linkId).get)
```

Comment:
Replacing `get` in the end with `getOrElse(Option.empty[RoadAddressLink])` would do the same thing as the `if-else if-else`, right?